docs-src/extensionFunctions.adoc

```diff
@@ -0,0 +1,20 @@
+=== Extension functions
+
+Extension functions allow us to attach custom functions to any library or custom code we write. You can extend String, Collection or 
+any other known type with some custom behavior. Given example creates `isContainedBy` extension function that we can directly call on 
+any kind of `T` type, as though it was part of the standard library.
```

Comment:
Looks good. Do you want to issue a new PR with the new format? I just updated master. We're now using markdown.